iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/module/AbstractBuilderFactory.java

```diff
@@ -0,0 +1,67 @@
+package eu.dnetlib.iis.wf.export.actionmanager.module;
+
+import static eu.dnetlib.iis.wf.export.actionmanager.ExportWorkflowRuntimeParameters.EXPORT_ALGORITHM_PROPERTY_SEPARATOR;
+import static eu.dnetlib.iis.wf.export.actionmanager.ExportWorkflowRuntimeParameters.EXPORT_TRUST_LEVEL_THRESHOLD;
+
+import org.apache.hadoop.conf.Configuration;
+
+import eu.dnetlib.iis.common.WorkflowRuntimeParameters;
+import eu.dnetlib.iis.common.hbase.HBaseConstants;
+
+/**
+ * Abstract action builder factory.
+ * 
+ * @author mhorst
+ *
+ * @param <T> avro input type
+ */
+public abstract class AbstractBuilderFactory<T> implements ActionBuilderFactory<T> {
+
+    /**
+     * Algorithm name associated with builder.
+     */
+    protected final AlgorithmName algorithmName;
+
+    // ------------------ CONSTRUCTORS -----------------------
+
+    AbstractBuilderFactory(AlgorithmName algorithmName) {
+        this.algorithmName = algorithmName;
+    }
+
+    // ------------------ LOGIC ------------------------------
+    
+    @Override
+    public AlgorithmName getAlgorithName() {
+        return algorithmName;
+    }
+
+    /**
+     * Provides trust level threshold if defined for given algorithm or globally.
+     * @param context
+     * @return trust level threshold or null if not defined
+     */
+    protected Float provideTrustLevelThreshold(Configuration conf) {
+        String algorithmTrustLevelThreshold = conf
+                .get(EXPORT_TRUST_LEVEL_THRESHOLD + EXPORT_ALGORITHM_PROPERTY_SEPARATOR + algorithmName.name());
+        if (algorithmTrustLevelThreshold != null
+                && !WorkflowRuntimeParameters.UNDEFINED_NONEMPTY_VALUE.equals(algorithmTrustLevelThreshold)) {
+            return Float.valueOf(algorithmTrustLevelThreshold);
+        }
+        String defaultTrustLevelThresholdStr = conf.get(EXPORT_TRUST_LEVEL_THRESHOLD);
+        if (defaultTrustLevelThresholdStr != null
+                && !WorkflowRuntimeParameters.UNDEFINED_NONEMPTY_VALUE.equals(defaultTrustLevelThresholdStr)) {
+            return Float.valueOf(defaultTrustLevelThresholdStr);
+        }
```

Comment:
`WorkflowRuntimeParameters` have useful method `getParamValue` for retrieving params that are aware of `$UNDEFINED` constant. I would use it (after changing signature to use `Configuration` object instead of `JobContext`).
I would even consider to extend `WorkflowRuntimeParameters` to handle this whole situation (as separate method):
"Param was not found so use default param as fallback". I am pretty sure that this kind of getting params is used in couple places in iis.